File: src/test/kotlin/TagsInputBoxTest.kt
import helpers.BaseTest
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import pages.TagsInputBoxPage

class TagsInputBoxTest : BaseTest() {

    @Test
    fun checkAddTag() {
        val maxTagsCount = TagsInputBoxPage()
            .open()
            .removeAllTags()
            .getTagsCount()

        val addedTags = TagsInputBoxPage()
            .addTag("Tag1")
            .addTag("Tag2")
            .getTags()

        val tagsCount = TagsInputBoxPage().getTagsCount()

        addedTags.firstOrNull { it.text == "Tag1" }.shouldNotBeNull()
        addedTags.firstOrNull { it.text == "Tag2" }.shouldNotBeNull()
        (maxTagsCount - addedTags.size) shouldBe tagsCount
    }
}

File: tests/tags_input_box.test.ts
import { test, expect } from '@playwright/test'
import { TagsInputBoxPage } from '../pages/TagsInputBoxPage'

test.describe('Tags input box', () => {
    let tagsInputBoxPage: TagsInputBoxPage

    test.beforeEach(async ({ page }) => {
        tagsInputBoxPage = new TagsInputBoxPage(page)
    
        await tagsInputBoxPage.open()
      })

    test('Check add tags', async ({ page }) => {
        let maxTagsCount = await (await tagsInputBoxPage
            .removeAllTags())
            .getTagsCount()
        console.log(maxTagsCount)

        let addedTags = await (await (await tagsInputBoxPage
            .addTag("Tag1"))
            .addTag("Tag2"))
            .getTags()
        console.log(addedTags)

        let tagsCount = await tagsInputBoxPage.getTagsCount()
        console.log(tagsCount)

        expect(addedTags.find(tag => tag.text === "Tag1")).toBeDefined()
        expect(addedTags.find(tag => tag.text === "Tag2")).toBeDefined()
        expect(tagsCount).toBe(maxTagsCount - addedTags.length)
  })
})